model/fn-execution/src/main/proto/org/apache/beam/model/fn_execution/v1/beam_fn_api.proto

```diff
@@ -46,6 +46,50 @@ import "google/protobuf/struct.proto";
 import "google/protobuf/timestamp.proto";
 import "google/protobuf/duration.proto";
 
+
+// Describes transforms necessary to execute Beam over the FnAPI but are
+// implementation details rather than part of the core model.
+message FnApiTransforms {
+  enum Runner {
+    // DataSource is a Root Transform, and a source of data for downstream
+    // transforms in the same ProcessBundleDescriptor. 
+    // It represents a logical PCollection.
```

Comment:
I might say "It represents reading a PCollection from an external source (typically the Runner)" or "represents a stream of values coming in from an external source/over a data channel" as it's not the PCollection itself, but rather a descriptor of how to get the PCollection.

Reply:
Done.